spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/support/ResponseEntityDecoder.java

```diff
@@ -0,0 +1,64 @@
+
+package org.springframework.cloud.netflix.feign.support;
+
+import feign.FeignException;
+import feign.Response;
+import feign.codec.Decoder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.LinkedList;
+
+/**
+ * Decoder adds compatibility for Spring MVC's ResponseEntity to any
+ * other decoder via composition.
+ * @author chadjaros
+ */
+@Slf4j
+public class ResponseEntityDecoder implements Decoder {
```

Comment:
I have the requirement to provide an API interface which can be used with or without Spring. I decided to not incorporate this into your SpringDecoder because it relies on spring autowiring. I ended up forking your feign codecs and wrapping a non-autowired version of them. I wanted the ReponseEntity decoding to be usable with any decoder, thus the composition based approach.